### PR TITLE
add nanoGPT

### PR DESCRIPTION
**Pull request by dkupnicki**

### CODE REVIEW

**Comment by jan-grzybek-ampere** [comment]
@MarcelWilnicki please test and review the NanoGPT code sometime this week, I will do the second pass

**Comment by jan-grzybek-ampere** [comment]
@MarcelWilnicki please have a look

File: utils/nlp/lambada.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import string
+from collections import Counter
+
+import utils.misc as utils
+import re
+
+class Lambada:
```
Comment (MarcelWilnicki) [inline utils/nlp/lambada.py:7]: 2 lines

File: utils/nlp/lambada.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        self.__exact_match_count = 0
+        self.__f1_count = 0
+
+    def __examples(self):
+        with open(self.__dataset, encoding="utf-8") as f:
+            for line in f:
+                yield line
+
+
+    def __load_next_inputs_maybe(self):
```
Comment (MarcelWilnicki) [inline utils/nlp/lambada.py:39]: "Method definitions inside a class are surrounded by a single blank line"

File: utils/nlp/lambada.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
+import string
+from collections import Counter
+
+import utils.misc as utils
+import re
```
Comment (MarcelWilnicki) [inline utils/nlp/lambada.py:5]: I suggest we keep builtin modules at the top, then third-party packages then our own modules inside aml. (each separated by two lines, applies to all files)

import re
import string
from collections import Counter

import utils.misc as utils

**Review by MarcelWilnicki — APPROVED** [review]
I approve, but consider addressing my formatting comments to stay inline with conventions.

**Comment by jan-grzybek-ampere** [comment]
@dkupnicki please ping me when ready for final review, thanks!

**Pull request merged**